sphinx_gallery/gen_rst.py

```diff
@@ -410,17 +400,23 @@ def scale_image(in_fname, out_fname, max_width, max_height):
                           generated images')
 
 
-def save_thumbnail(thumbnail_image_path, base_image_name, gallery_conf):
+def save_thumbnail(thumbnail_image_path, src_file, gallery_conf):
     """Save the thumbnail image"""
     thumb_dir = os.path.join(os.path.dirname(thumbnail_image_path), 'thumb')
     if not os.path.exists(thumb_dir):
         os.makedirs(thumb_dir)
 
+    base_image_name = os.path.splitext(os.path.basename(src_file))[0]
     thumb_file = os.path.join(thumb_dir,
                               'sphx_glr_%s_thumb.png' % base_image_name)
 
-    if os.path.exists(thumbnail_image_path):
+    if src_file in gallery_conf['failing_examples']:
```

Comment:
Maybe put a plot before the error in plot_raise.py so that we can check that even if one plot is produced it doesn't find its way to the gallery. Something like:
```diff
diff --git a/examples/plot_raise.py b/examples/plot_raise.py
index db504a8..1fe5209 100644
--- a/examples/plot_raise.py
+++ b/examples/plot_raise.py
@@ -9,6 +9,12 @@ image stamp. Thus allowing easy identification in the gallery display.
 
 You also get the python traceback of the failed code block
 """
+import numpy as np
+import matplotlib.pyplot as plt
+
+plt.pcolormesh(np.random.randn(100, 100))
+
+###############################################################################
 
 iae
```